functions/Get-DbaDbLogSpace.ps1

```diff
@@ -0,0 +1,139 @@
+
+        Returns the transaction log usage information for all databases on Server1, except the system databases
+
+    .EXAMPLE
+        PS C:\> Get-DbaRegisteredServer -SqlInstance cmsServer | Get-DbaDbLogSpace -Database Database1
+
+        Returns the transaction log usage information for Database1 for a group of servers from SQL Server Central Management Server (CMS).
+
+    #>
+    [CmdletBinding()]
+    param (
+        [parameter(ValueFromPipeline, Mandatory)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Database,
+        [string[]]$ExcludeDatabase,
+        [switch]$ExcludeSystemDatabase,
+        [switch]$EnableException
+    )
+
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+            } catch {
+                Stop-Function -Message "Failed to process Instance $Instance" -ErrorRecord $_ -Target $instance -Continue
+            }
+            $dbs = $server.Databases | Where-Object IsAccessible
+
+            if ($Database) {
+                $dbs = $dbs | Where-Object Name -in $Database
+            }
+            if ($ExcludeDatabase) {
+                $dbs = $dbs | Where-Object Name -NotIn $ExcludeDatabase
+            }
+
+            if ($ExcludeSystemDatabase) {
+                $dbs = $dbs | Where-Object IsSystemObject -eq $false
+            }
+
+            # 2012+ use new DMV
+            if ($server.versionMajor -ge 11) {
+                foreach ($db in $dbs) {
+                    try {
+                        $logspace = $server.query('select * from sys.dm_db_log_space_usage', $db.name)
+                    } catch {
+                        Stop-Function -Message "Unable to select from sys.dm_db_log_space_usage on $instance." -ErrorRecord $_ -Target $db -Continue
+                    }
+                    [pscustomobject]@{
+                        ComputerName        = $server.ComputerName
+                        InstanceName        = $server.ServiceName
+                        SqlInstance         = $server.DomainInstanceName
+                        Database            = $db.name
+                        LogSize             = [dbasize]($logspace.total_log_size_in_bytes)
+                        LogSpaceUsedPercent = $logspace.used_log_space_in_percent
+                        LogSpaceUsed        = [dbasize]($logspace.used_log_space_in_bytes)
+                    }
+                }
+            } else {
+                write-host 'under 2012'
+                try {
+                    $logspace = $server.Query("dbcc sqlperf(logspace)") | Where-Object { $dbs.name -contains $_.'Database Name' }
+                } catch {
+                    Stop-Function -Message "Unable to run dbcc sqlperf on $instance." -ErrorRecord $_ -Target $db -Continue
+                }
```

Comment:
I don't think we need a try/catch for both with a custom message. Think we can just stick with a generic message as we do on other commands like `Unable to collect log space data on $instance`